### PR TITLE
[jk] Edit bookmark values

### PR DESCRIPTION
**Pull request by johnson-mage**
# Summary
- Edit bookmark property values from UI. User can edit the bookmark values, which will be used as a bookmark for the next sync. The bookmark values will automatically update to the last record synced after the next sync is completed.
- Made backend API changes to handle requests from FE.

# Tests
edit bookmark values

### CODE REVIEW

**Comment by johnson-mage** [comment]
@dy46 Can you please review my BE changes again? Thanks!

File: mage_ai/data_preparation/models/block/__init__.py [hunk 1]
```diff
@@ -1338,6 +1355,31 @@ def update(self, data):
             if self.has_callback:
                 CallbackBlock.create(self.uuid)
             self.__update_pipeline_block()
+        if (
```
Comment (dy46) [inline mage_ai/data_preparation/models/block/__init__.py:1358]: you can add some of this integration pipeline specific logic in the `IntegrationBlock` class

File: mage_ai/data_preparation/models/block/integration/__init__.py [hunk 2]
```diff
@@ -314,6 +314,30 @@ class SourceBlock(IntegrationBlock):
 
 
 class DestinationBlock(IntegrationBlock):
+    def update(self, data, update_state=False):
```
Comment (johnson-mage) [inline mage_ai/data_preparation/models/block/integration/__init__.py:317]: @dy46 Ok i moved the update logic to the `DestinationBlock` specifically cuz it's for updating the destination state  in particular.

File: mage_ai/data_preparation/models/block/__init__.py [hunk 3]
```diff
@@ -1280,6 +1282,21 @@ def to_dict(
...
+                destination_table=destination_table,
+                stream=state_stream,
+            )
+            if os.path.isfile(destination_state_file_path):
+                with open(destination_state_file_path, 'r') as f:
+                    text = f.read()
+                    d = json.loads(text) if text else {}
+                    bookmark_values = d.get('bookmarks', {}).get(state_stream)
+                    data['bookmarks'] = bookmark_values
+
```
Comment (dy46) [inline mage_ai/data_preparation/models/block/__init__.py:1299]: i would also move this out

Comment (johnson-mage) [reply mage_ai/data_preparation/models/block/__init__.py:1299]: oh gotcha

**Review by dy46 — APPROVED** [review]

**Pull request merged**